Send email when process success and failed

anshour opened the pull request:
https://inisiatif-zakat.atlassian.net/browse/PHP-536

anshour committed:
Add psr-4 autoload for factories

anshour committed:
Add some default config and migrations

anshour committed:
Add employee id

anshour committed:
Add some factories

anshour committed:
Add recording migrations for testing only

anshour committed:
Add mail method

anshour committed:
Add employee relation

anshour committed:
Change mail view path

anshour committed:
Add employee id attribute

anshour committed:
Add sending recap status job

anshour committed:
Move and rename mail template

anshour committed:
Add mail for success and failed recap process

anshour committed:
Add test for sending success and failed mail

nuradiyana commented on database/migrations/create_donation_donor_recap_table.php:
```diff
@@ -22,6 +22,11 @@ public function up(): void
 
         Schema::create('donation_recaps', static function (Blueprint $table): void {
             $table->uuid('id')->primary();
+            $table->foreignUuid('employee_id')
+                ->nullable()
+                ->constrained('employees')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
```
Ini tambah table berarti ya?

nuradiyana replied:
Pakai `user_id` aja keknya, ga perlu employee

nuradiyana commented on src/Jobs/ProcessDonationRecap.php:
```diff
@@ -36,6 +36,8 @@ public function handle(): void
             new IncreaseProgressDonationRecap($this->donationRecap),
             new CheckDonationRecapProgress($this->donationRecap),
         ]));
+
+        \dispatch(new SendingRecapStatusJob($this->donationRecap));
```
Ini keknya panggil di `CheckDonationRecapProgress` ya? karena kan di sana ubah status jadi error atau done?

anshour committed:
Add employee id

anshour committed:
Add single boolean column

anshour committed:
Change mail status template

anshour committed:
Add employee id in recap data

anshour committed:
Change increase progress logic and handle email

anshour committed:
Removing test because useless

anshour committed:
Update sending recap status job

anshour committed:
Change employee to user relation recap

Add user table config

Remove employee factory (not used)

Change employee to user id column

Change employee id to user id

Change employee to user

Add model user

Add user table name

anshour committed:
Add user table name

anshour committed:
Change employee relation to user

anshour committed:
Revert "Change employee to user relation recap"

This reverts commit 1cad7d0d3cd9def8a2b40ddcbd0b8fe86a39bf63.

anshour committed:
Revert "Change employee relation to user"

This reverts commit 555b3d214977277afc942082f8788fe9c91db97c.

anshour committed:
Revert "Add user table name"

This reverts commit 9acf6ec693d35a4644a441e8e60b9de3c910a993.
